common/serde-reflection/src/de.rs

```diff
@@ -0,0 +1,604 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    error::{Error, Result},
+    format::{ContainerFormat, ContainerFormatEntry, Format, FormatHolder, Named, VariantFormat},
+    trace::Tracer,
+};
+use serde::de::{self, DeserializeSeed, IntoDeserializer, Visitor};
+use std::collections::BTreeMap;
+
+pub(crate) struct Deserializer<'a> {
+    tracer: &'a mut Tracer,
+    format: &'a mut Format,
+}
+
+impl<'a> Deserializer<'a> {
+    pub(crate) fn new(tracer: &'a mut Tracer, format: &'a mut Format) -> Self {
+        Deserializer { tracer, format }
+    }
+}
+
+impl<'de, 'a> de::Deserializer<'de> for Deserializer<'a> {
+    type Error = Error;
+
+    fn deserialize_any<V>(self, _visitor: V) -> Result<V::Value>
+    where
+        V: Visitor<'de>,
+    {
+        Err(Error::NotSupported("deserialize_any"))
+    }
+
+    fn deserialize_bool<V>(self, visitor: V) -> Result<V::Value>
+    where
+        V: Visitor<'de>,
+    {
+        self.format.merge(Format::Bool)?;
```

Comment:
I think I need a little more context (documentation) about what the format functions do (merge/push, etc) and how  they're being used. Probably because my brain is a little fried after going through all this :)

Reply:
That's fair. It's also missing unit tests TBH.